packages/lesswrong/lib/collections/posts/schema.tsx

```diff
@@ -133,6 +135,15 @@ const schemaDefaultValueFmCrosspost = schemaDefaultValue({
   isCrosspost: false,
 })
 
+const sequenceIdIfPermitted = async (post, currentUser) => {
+  if (!post.canonicalSequenceId) return null;
+
+  const sequence = await Sequences.findOne({ _id: post.canonicalSequenceId });
+  if (!Sequences.options.mutations.edit.check(currentUser, sequence)) return null;
```

Comment:
Given the extremely infrequent use of this pattern (which I didn't even realize existed until now), I think I'd probably prefer to avoid adding more of it to the codebase.  (Due to poor typing, it's difficult to figure out what it's actually doing.)

If the `sequence` collection's `editCheck` is appropriate for both `onCreate` and `onUpdate` here, though, I'd maybe just pull the check out to its own function and use it both here and in the collection declaration.  Functionally I think this works out to the same thing but is much easier to dig into since it doesn't depend on functionality which was dynamically generated at runtime.

If I'm misunderstanding how this works please let me know.

Reply:
I've changed this so that Sequences now has a checkEditAccess function, similar to the checkAccess function that all collections have, and I used that both here and in the editCheck function in sequences/collection.ts (which I think is what you meant with "in the collection declaration" but I'm not very familiar with the codebase so might be mistaken).